Solutions

Ngayka opened the pull request.

Ngayka committed:
Solutions

mateacademy-ai-mentor commented on cinema_service/settings.py:
```diff
@@ -101,16 +125,13 @@
     {
-        "NAME": "django.contrib.auth.password_validation."
-        "MinimumLengthValidator",
+        "NAME": "django.contrib.auth.password_validation." "MinimumLengthValidator",
```
Issue: The 'NAME' value for the password validator is a concatenated string: 'django.contrib.auth.password_validation.' 'MinimumLengthValidator'. It should be a single string: 'django.contrib.auth.password_validation.MinimumLengthValidator'. Please update this and similar entries for proper Django configuration.

mateacademy-ai-mentor commented on cinema_service/settings.py:
```diff
@@ -101,16 +125,13 @@
     {
-        "NAME": "django.contrib.auth.password_validation."
-        "CommonPasswordValidator",
+        "NAME": "django.contrib.auth.password_validation." "CommonPasswordValidator",
```
Issue: The 'NAME' value for the password validator is a concatenated string: 'django.contrib.auth.password_validation.' 'CommonPasswordValidator'. It should be a single string: 'django.contrib.auth.password_validation.CommonPasswordValidator'.

mateacademy-ai-mentor commented on cinema_service/settings.py:
```diff
@@ -101,16 +125,13 @@
     {
-        "NAME": "django.contrib.auth.password_validation."
-        "NumericPasswordValidator",
+        "NAME": "django.contrib.auth.password_validation." "NumericPasswordValidator",
```
Issue: The 'NAME' value for the password validator is a concatenated string: 'django.contrib.auth.password_validation.' 'NumericPasswordValidator'. It should be a single string: 'django.contrib.auth.password_validation.NumericPasswordValidator'.

mateacademy-ai-mentor reviewed (changes requested):
### Review Summary

Thank you for your hard work! Before this PR can be approved, please address the critical missing requirements: Swagger documentation (including query parameter documentation for filtering) and throttling settings for both unauthorized and authorized users. Additionally, please correct the 'NAME' values for password validators in your settings.py to be single strings, as this is required for proper Django configuration. Once these core issues are resolved, your project will be much closer to completion! 🚀

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)

> [!IMPORTANT]
> Fix the issues, commit the changes, and then re-request my review using the **`Re-request review` button**[^2] in the Reviewers tab.

[^2]: [Re-request AI review example](https://snipboard.io/nOXxHN.jpg)

> [!TIP]
> If you have any comments about this review, please leave a message in the [Add a comment](#new_comment_field) section below and **mention me with the `@mateacademy-ai-mentor` tag**[^3].
> I will read your message during my next review of your pull request. Please do not reply directly to my comments, as that feature is not supported yet.

[^3]: [Add AI review comments example](https://snipboard.io/hVlOey.jpg)

> [!CAUTION]
> Do not close the pull request until the review is complete. Closing it early can lead to unexpected behavior and disrupt the review process.

Ngayka committed:
Fix password validator config, add throttling & query param docs (Swagger)

- Fixed incorrect 'NAME' values in AUTH_PASSWORD_VALIDATORS (strings were split across lines)
- Added proper throttling config to REST_FRAMEWORK:
    - 10 requests/minute for anonymous users
    - 30 requests/minute for authenticated users
- Created `cinema/schemas.py` with reusable OpenApiParameter objects for filtering
- Replaced hardcoded Swagger query parameter definitions in MovieViewSet and MovieSessionViewSet with reusable ones from `schemas.py`
- Changed all `OpenApiParameter.location` to string literals (e.g. "query") to fix type warnings
- Minor refactoring for clarity and Swagger schema maintainability

mateacademy-ai-mentor commented on cinema/schemas.py:
```diff
@@ -0,0 +1,31 @@
+    name="movie",
+    type=OpenApiTypes.INT,
+    location="query",
+    description="Filter by movies id (ex. ?movie=1,2", )
```
Minor issue: The description for the 'movie' filter parameter is missing a closing parenthesis in the example (`ex. ?movie=1,2`). Consider updating it to match the format of the other descriptions for consistency and clarity.

mateacademy-ai-mentor reviewed (approved):
### Review Summary

Great job! 🎉 All core requirements have been fully met, including Swagger documentation, throttling, JWT support, and comprehensive tests. The only issue found is a minor typo in the 'movie' filter parameter description (missing a closing parenthesis), which does not affect functionality. Consider fixing this for clarity, but your code is ready to be merged. Keep up the excellent work!

***

✨ Please rate this review in the Platform's notification center[^1]. I will be grateful for your feedback! ✨

[^1]: [Rate AI review example](https://snipboard.io/gD6XKF.jpg)
